File: 0024_Swap_Nodes_in_Pairs.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* swapPairs(ListNode* head) {
        ListNode *t = NULL, *origin = new ListNode(0), *prev = origin;
        origin->next = head;
        while (head != NULL && head->next != NULL) {
            t = head->next;
            prev->next = t;
            head->next = head->next->next;
            t->next = head;
            prev = head;
            head = head->next;
        }
        return origin->next;
    }
};

File: 0024_Swap_Nodes_in_Pairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(next=head)
        ref = dummy
        while ref.next and ref.next.next:
            a, b = ref.next, ref.next.next
            temp = b.next
            
            ref.next = b
            b.next = a
            a.next = temp
            
            ref = a
        return dummy.next
